src/tokenizer/ngram_tokenizer.rs

```diff
@@ -74,41 +146,21 @@ impl<'a> TokenStream for NgramTokenStream<'a> {
     // clear out working token text
     self.token.text.clear();
 
-    loop {
-      // can we proceed?
-      if (self.location + self.gram_size) <= self.text_length {
-        // this token's position is our current location
-        self.token.position = self.location;
-
-        // cycle through 1,2 (min...max)
-        match self.cycle() {
-          Some(size) => {
-            let offset_from = self.location;
-            let offset_to = offset_from + size;
-
-            self.token.offset_from = offset_from;
-            self.token.offset_to = offset_to;
-
-            self.token.text.push_str(&self.text[offset_from..offset_to]);
-          }
-          None => {
-            if self.edges_only {
-              break;
-            }
-
-            // move us down the chain of letters
-            self.location = self.location + 1;
-            continue;
-          }
-        }
-
-        return true;
-      } else {
-        break;
-      }
-    }
+    if let Some((location, size)) = self.chomp() {
+      // this token's position is our current location
```

Comment:
let's  not use the word `location`. `position` or `offset`

Reply:
Fixed